docs/design.html

```diff
@@ -353,3 +353,35 @@
   <li>You cannot configure yet how much log is retained without compaction (the "head" of the log).  Currently all segments are eligible except for the last segment, i.e. the one currently being written to.</li>
   <li>Log compaction is not yet compatible with compressed topics.</li>
 </ol>
+<h3><a id="semantics">4.9 Quotas</a></h3>
+<p>
+    Starting in 0.9, the Kafka cluster has the ability to enforce quotas on produce and fetch requests. Quotas are basically byte-rate thresholds defined per client-id. A client-id logically identifies an application making a request. Hence a single client-id can span multiple producer and consumer instances and the quota will apply for all of them as a single entity i.e. if client-id="test-client" has a produce quota of 10MB/sec, this is shared across all instances with that same id.
+
+<h4>Why are quotas necessary?</h4>
+<p>
+It is possible for producers and consumers to produce/consume very high volumes of data and thus monopolize broker resources, cause network saturation and generally DOS other clients and the brokers themselves. Having quotas protects against these issues and is all tbe more important in large multi-tenant clusters where a small set of badly behaved clients can degrade user experience for the well behaved ones. In fact, when running Kafka as a service this even makes it possible to enforce API limits according to an agreed upon contract.
+</p>
+<h4>Enforcement</h4>
+<p>
+    By default, each unique client-id receives a fixed quota in bytes/sec as configured by the cluster (quota.producer.default, quota.consumer.default).
+    This quota is defined on a per-broker basis. Each client can publish/fetch a maximum of X bytes/sec per broker before it gets throttled. We decided that defining these quotas per broker is much better than having a fixed cluster wide bandwidth per client because that would require a mechanism to share client quota usage among all the brokers. This can be harder to get right than the quota implementation itself!
+</p>
+<p>
+    How does the Kafka Server react when it detects a quota violation? We have a few options:
+    <ol>
+      <li>Return an error immediately</li>
+      <li>Delay the response to slow down the guilty clients</li>
+    </ol>
+
+    We chose option 2 in our solution. The broker does not return an error rather it attempts to slow down a client exceeding it's quota. It computes the amount of delay
+needed to bring the guilty client under it's quota and delays sending the response for that time. This approach is better than option 1 primarily because this keeps the quota violation transparent to clients (outside of client side metrics). Clients don't need to implement any special backoff and retry behavior. In addition, bad client behavior (retry without backoff) can exacerbate the very problem quotas are trying to solve.
```

Comment:
Good point. My description made it look like an available option. I actually included it because there are some pieces in the design section that discuss tradeoffs (for e.g. Push v Pull). But those are more fundamental in nature so I suppose they are fine.

Let me know if the new version is more concise.